Read version info from `nf-core/demultiplex` version file

Aratz opened the pull request:
This PR adds support for runfolders processed with [nf-core/demultiplex](https://github.com/nf-core/demultiplex/) instead of [arteria-bcl2fastq](https://github.com/arteria-project/arteria-bcl2fastq/).

In the current version of snpseq_packs, the version of bcl2fastq is dumped to a file during processing. This won't be needed in the future since all nf-core pipelines collect all software versions in a single file.

Aratz committed:
Add nf-core/demultiplex version parser

Aratz committed:
Apply black

matrulda commented on bin/get_metadata.py:
```diff
@@ -82,6 +83,14 @@ def get_bcl2fastq_version(self, runfolder):
+            return dict(
+                line.strip().split(": ")
+                for line in f.readlines()
+                if line.startswith("  ")
```
Ah smart! But I wonder if this could be dangerous if the yaml is formatted in a different way? i.e the indent is another number of spaces. I guess this is a nf-core standard so the risk is probably small, but still. Could potentially be safer to import it with a yaml package and handle the values of all top-level keys. Let me know what you think.

Aratz replied:
Good point, fixed it now :+1:

matrulda reviewed (commented):
Looks good! Just had a comment that you can have a look at.

matrulda commented on tests/unit_tests/test_get_metadata.py:
```diff
@@ -58,6 +58,20 @@ def test_bcl2fastq_version(runfolder_info):
+        "bcl2fastq": "2.20.0.422",
+        "python": "3.12.0",
+        "yaml": "6.0.1",
+        "Nextflow": "23.04.3",
```
I realized another thing: We should make sure it is clear that these are nf-core/Demultiplex versions. We might use other versions of Nextflow and python in the seqreports pipeline. Alternatively, only use keys: `bcl2fastq` and `nf-core/demultiplex`.

Aratz replied:
Also fixed, all versions related to demultiplexing are now gathered in a "Demultiplexing" subsection

Aratz committed:
Parse yaml file with yaml module

Aratz committed:
Reorganize version listing

This is to make it clear that these version come from e.g. the
nf-core/demultiplex pipeline and not the seqreports pipeline.

Aratz committed:
Run Black

Aratz commented:
Thanks for the review! Should be fixed now :)

matrulda commented on bin/get_metadata.py:
```diff
@@ -82,6 +84,14 @@ def get_bcl2fastq_version(self, runfolder):
+            return {
+                software: version
+                for sotfware_dict in yaml.safe_load(f).values()
+                for software, version in sotfware_dict.items()
```
Super minor: Typo in variable name `sotfware_dict` ? Might be nice to fix.

Aratz replied:
Good catch! This should have been caught by the tests though :thinking: , will investigate and fix!

Aratz replied:
Oh I see, the typo was for all the instances of this variables :sweat_smile:

matrulda reviewed (approved):
Great! Just a super minor comment

Aratz committed:
Fix typo
